pkg/kubefed2/federate/federate.go

```diff
@@ -124,50 +134,54 @@ func NewCmdFederateResource(cmdOut io.Writer, config util.FedConfig) *cobra.Comm
 	return cmd
 }
 
+type FedResources struct {
+	*enable.TypeResources
+	FedResource *unstructured.Unstructured
+}
+
 // Run is the implementation of the `federate resource` command.
 func (j *federateResource) Run(cmdOut io.Writer, config util.FedConfig) error {
 	hostConfig, err := config.HostConfig(j.HostClusterContext, j.Kubeconfig)
 	if err != nil {
 		return errors.Wrap(err, "Failed to get host cluster config")
 	}
 
-	qualifiedTypeName := ctlutil.QualifiedName{
-		Namespace: j.FederationNamespace,
-		Name:      j.typeName,
-	}
-
 	qualifiedResourceName := ctlutil.QualifiedName{
 		Namespace: j.resourceNamespace,
 		Name:      j.resourceName,
 	}
 
-	resources, err := GetFedResources(hostConfig, qualifiedTypeName, qualifiedResourceName, j.outputYAML)
+	typeResources, err := getTypeResources(hostConfig, j.typeName, j.FederationNamespace, j.enableType, j.outputYAML)
+	if err != nil {
+		return err
+	}
+	fedResource, err := GetFedResource(hostConfig, typeResources.TypeConfig, qualifiedResourceName)
 	if err != nil {
 		return err
 	}
 
+	fedResources := &FedResources{
+		FedResource:   fedResource,
+		TypeResources: typeResources,
+	}
+
 	if j.outputYAML {
-		err := util.WriteUnstructuredToYaml(resources.FedResource, cmdOut)
+		objs := []interface{}{}
+		if j.enableType {
```

Comment:
Why should --enable-type be respected if yaml output is requested?

Reply:
I did initially think of an use case that a user might want to collect all yamls, before creating any of them. Revisiting the thought, this might become problematic (unless we want to make `enable type` idempotent) if many resources of the same type needs to be federated, in which case the type yamls will be repeated. The updated implementation considers `--enable-type` and `-o yaml` mutually exclusive.